backend/src/main/java/ai/verta/modeldb/artifactStore/storageservice/s3/S3Client.java

```diff
@@ -112,7 +112,17 @@ private void initializeWithTemporaryCredentials(Regions awsRegion)
     AWSSecurityTokenService stsClient = null;
     AmazonS3 newS3Client = null;
     try {
-      stsClient = AWSSecurityTokenServiceClientBuilder.standard().withRegion(awsRegion).build();
+      final var stsClientBuilder =
+          AWSSecurityTokenServiceClientBuilder.standard().withRegion(awsRegion);
+      if (!CommonUtils.appendOptionalTelepresencePath("foo").equals("foo")) {
+        stsClientBuilder.setCredentials(
+            WebIdentityTokenCredentialsProvider.builder()
+                .webIdentityTokenFile(
+                    CommonUtils.appendOptionalTelepresencePath(
+                        System.getenv(ModelDBConstants.AWS_WEB_IDENTITY_TOKEN_FILE)))
+                .build());
+      }
+      stsClient = stsClientBuilder.build();
```

Comment:
What is the reason behind this?

Reply:
It was in a separate PR (see this comment: https://github.com/VertaAI/modeldb/pull/2869/files#r810393587) - that PR has been merged into main so I'm not sure why those commits are showing up in this PR.